backend/src/common/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,38 @@
+import fp from 'fastify-plugin';
+
+import {
+    ControllerHook,
+    ExceptionMessage,
+    HttpCode,
+    HttpError,
+} from '../../enums/enums.js';
+
+const authorization = fp(async (fastify, { routesWhiteList, services }) => {
+    fastify.decorateRequest('user', null);
+
+    fastify.addHook(ControllerHook.ON_REQUEST, async (request, reply) => {
+        try {
+            const isWhiteRoute = routesWhiteList.includes(request.routerPath);
+
+            if (isWhiteRoute) {
+                return;
+            }
+
+            const [, token] = request.headers.authorization?.split(' ') ?? [];
```

Comment:
looks like you can use `getToken` helper

Reply:
Resolved in [5e998f7](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/115/commits/5e998f7ffdb31243b124092254855ece5e7c383a)